www/src/components/graphiql-embed.js

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useRef, useCallback, useLayoutEffect } from "react"
+
+const VIDEO_RATIO = 2 / 3
+
+const GraphiQLEmbed = ({ title, url, query }) => {
+  const encodedQuery = encodeURIComponent(query)
+
+  const [iframeWidth, setIframeWidth] = useState(0)
+  const iframeRef = useRef()
+
+  const handleResize = useCallback(
+    () => setIframeWidth(iframeRef.current.clientWidth),
+    [iframeRef.current]
+  )
+
+  useLayoutEffect(() => {
+    handleResize()
+    window.addEventListener(`resize`, handleResize)
+
+    return () => {
+      window.removeEventListener(`resize`, handleResize)
+    }
+  }, [iframeRef.current])
+
+  return (
+    <iframe
```

Comment:
just curious, is iframe the preferred strategy here? if not, is there a way i can help with a more preferrable solution to this?

Reply:
@acao did you have something in mind?

Reply:
Well, if the gatsby-plugin-graphiql doesn’t already export a react component encapsulation, we could refactor it to do so, and we could render it here

Reply:
if that’s useful and more advantageous than the iframe, I can make the pr for the plugin that would enable you on this side

Reply:
I don't think so. We don't want the graphiql for the website itself, we want it for a sample project hosted on CodeSandbox.

Reply:
oh ok! well, despite it being able to run in codesandbox just fine, i see what you mean. to note for the future, GraphiQL need not fetch schema or execute queries over HTTP, if that's ever helpful, but that's more something to mention to the core team